SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs

```diff
@@ -58,9 +58,13 @@ internal static AppCenter Instance
         {
             get
             {
+                if (_instanceField != null)
+                {
+                    return _instanceField;
+                }
                 lock (AppCenterLock)
                 {
-                    return _instanceField ?? (_instanceField = new AppCenter());
+                    return _instanceField = new AppCenter();
```

Comment:
Please check https://help.semmle.com/wiki/display/CSHARP/Double-checked+lock+is+not+thread-safe and add volatile and a second if check.

Reply:
Or maybe we can use `Lazy`.

Reply:
Yeah agree, I thought about second check, and missed it yesterday